### DISCUSSION TITLE
Feature Request - About clipping tables in web pages

### DISCUSSION THREAD

**Discussion by ClaudeH1024** [coddingtonbear/obsidian-web#93] [Ideas]
Whenever I clipped a page, the table in the page was not rendered in markdown table syntax in my notes. I'm not sure if this is a bug or an undeveloped feature, so I'm sorry for making this request here.

**Comment by coddingtonbear** [comment]
I'm afraid we're beholden to what features [Mozilla Readability](https://github.com/mozilla/readability) provides; so such a feature would need to be added there rather than here.

**Comment by cmkm** [comment]
@coddingtonbear Hi there, I'm one of the Readability maintainers. Would it be possible to provide more context on what functionality you'd need from Readability? [mozilla/readability#838](https://github.com/mozilla/readability/issues/838)

**Reply by j-peeters** [reply]
I came here to figure out why I couldn't capture tables. It would be great if it could become clear what is needed for readability view to be able to be captured by your extension @coddingtonbear

**Reply by coddingtonbear** [reply]
The rough process that pages go through when capturing a snapshot is that:

1. We capture the page content as HTML.
2. We pass that HTML content through Readability (see https://github.com/coddingtonbear/obsidian-web/blob/03e30e3b43b3e40b488c81705fb0a54b32e16c8e/src/popup.tsx#L480 and the contents of the function being called)
3. We convert the output from Readability into markdown via Turndown (via https://github.com/coddingtonbear/obsidian-web/blob/03e30e3b43b3e40b488c81705fb0a54b32e16c8e/src/popup.tsx#L497 and the contents of the function being called)
4. We provide that markdown to your template as a template variable.

My belief right now, given a glance at issues on Readability's issue tracker, is that the problem here is that in step (2) above tables are often lost.  It's totally possible the tables are lost at step (3), though, too. Unfortunately, without having even a single example URL above, and a limited amount of time to spend investigating bugs without concrete, reproducible reproduction steps, I can really only guess where the problem might lie.

**Reply by j-peeters** [reply]
Thanks @coddingtonbear for the write-up. One URL where I experience this is https://en.wikipedia.org/wiki/Hungarian_language. And I suppose all Wikipedia pages with tables.
